src/quacc/recipes/espresso/_base.py

```diff
@@ -218,4 +223,14 @@ def _prepare_atoms(
         **calc_flags,
     )
 
-    return atoms
+    if isinstance(copy_files, (str, Path, list)):
+        exact_files_to_copy = prepare_copy_files(calc.user_calc_params, binary=binary)
+
+        if isinstance(copy_files, (str, Path)):
+            copy_files = {copy_files: exact_files_to_copy}
+        elif isinstance(copy_files, list):
+            copy_files = {source: exact_files_to_copy for source in copy_files}
```

Comment:
This block can likely be improved without the need for so many instance checks.

Perhaps something like:

```python
if copy_files:
    if isinstance(copy_files, (str, Path)):
        copy_files = [copy_files]
    exact_files_to_copy = prepare_copy_files(calc.user_calc_params, binary=binary)
    copy_files = {source: exact_files_to_copy for source in copy_files}
```

That said, it might be helpful to not overwrite the `copy_files` variable, e.g. to return `updated_copy_files` or something similar, just for clarity.